### PR TITLE
ssh/tailssh: replace incubator process with su instead of running su as child

### PR DESCRIPTION
**Pull request by oxtoacart**
This allows the SSH_AUTH_SOCK environment variable to work inside of
su and agent forwarding to succeed.

Fixes https://github.com/tailscale/tailscale/issues/12467

### CODE REVIEW

File: ssh/tailssh/incubator.go [hunk 1]
```diff
@@ -438,8 +438,7 @@ func trySU(dlogf logger.Logf, ia incubatorArgs) (handled bool, err error) {
 	}
 
 	dlogf("logging in with %s %q", su, loginArgs)
-	cmd := newCommand(ia.hasTTY, su, loginArgs)
-	return true, cmd.Run()
+	return true, unix.Exec(su, loginArgs, os.Environ())
```
Comment (bradfitz) [inline ssh/tailssh/incubator.go:441]: hah. unix.Exec hiding over in a secondary result and relying on its result value on failure only is a hiding it a bit too much IMO

I'd pull it up a bit to make a big deal out of the exec and its err, like

```go
     dlogf("logging in with %s %q", su, loginArgs)

     // In the successful case, we never make it past this:
     err := unix.Exec(su, loginArgs, os.Environ())
    
     // If we made it this far, the exec failed.
     return true, err
```

**Review by bradfitz — COMMENTED** [review]
Does the unix.Exec

File: .github/workflows/ssh-integrationtest.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Run the ssh integration tests with `make sshintegrationtest`.
```
Comment (oxtoacart) [inline .github/workflows/ssh-integrationtest.yml:1]: While I'm in here, I'm adding the integration tests that I've run manually to CI. They'll only run if someone touches the SSH code.

File: ssh/tailssh/incubator.go [hunk 3]
```diff
@@ -438,8 +442,12 @@ func trySU(dlogf logger.Logf, ia incubatorArgs) (handled bool, err error) {
 	}
 
 	dlogf("logging in with %s %q", su, loginArgs)
-	cmd := newCommand(ia.hasTTY, su, loginArgs)
-	return true, cmd.Run()
+
+	// If Exec works, the Go code will not proceed past this:
+	err = unix.Exec(su, loginArgs, os.Environ())
```
Comment (oxtoacart) [inline ssh/tailssh/incubator.go:447]: The heart of the fix.

File: ssh/tailssh/tailssh_integration_test.go [hunk 4]
```diff
@@ -443,33 +568,23 @@ type testBackend struct {
 
 func (tb *testBackend) GetSSH_HostKeys() ([]gossh.Signer, error) {
 	var result []gossh.Signer
-	for _, typ := range []string{"ed25519", "ecdsa", "rsa"} {
```
Comment (oxtoacart) [inline ssh/tailssh/tailssh_integration_test.go:446]: ssh in the docker container only seems to like rsa.

**Review by bradfitz — APPROVED** [review]

**Pull request merged**